### ISSUE TITLE
Severe gif encoding bug

### ISSUE THREAD

**Issue by mjphaynes** [stil/gif-endec#15]
There is a bug here: https://github.com/stil/gif-endec/blob/9b9b7d4957a2283bdbd25d8a667a41c419c125f5/src/Encoder.php#L183

See here for correct logic: https://github.com/lunakid/AnimGif/blob/master/src/GifCreator/AnimGif.php#L361

This causes gifs to be corrupted in certain situations

**Comment by marrone** [comment]
I have an issue with corrupted gifs as well, but the linked code looks comparable. I think the issue is elsewhere, but I've not found it yet

**Comment by mjphaynes** [comment]
Yeah this could be your issue as well as this is a pretty major flaw in the logic.

I don't know if this package is still maintained? If it is I could submit a pull request?
